### PR TITLE
Implement header design

### PR DESCRIPTION
**Pull request by jaredkhan**
Implements the design that we've been carving out for the header of the landing page:

screen shot 2017-08-20 at 20 42 44

### CODE REVIEW

File: static/src/styles/landing/LandingIntroduction.styl [hunk 1]
```diff
@@ -1,3 +1,30 @@
...
+  color: $Color__mediumText;
+}
+
+.LandingIntroduction__link {
+  text-decoration: none;
+  color: $Color__brightText;
+}
+
+.LandingIntroduction__link:after {
+    content:' ➝';
```
Comment (Pinpickle) [inline static/src/styles/landing/LandingIntroduction.styl]: Now this is some semantic beauty right here 😍

File: Views/home.stencil [hunk 2]
```diff
@@ -3,15 +3,24 @@
...
-  </header>
-  <section class="LandingFeature">
-    <h2>Featured</h2>
-    <p>Hack Cambridge</p>
-  </section>
+  <div class="LandingTop">
+    <header class="LandingIntroduction LandingTop__Introduction">
+      <div class="LandingIntroduction__titleContainer">
+        <img src="static/images/hac-logo-dark.svg" alt="Hackers at Cambridge" class="LandingIntroduction__titleImage">
+        <h4 class="LandingIntroduction__tagLine">Cambridge's student tech society</h4>
```
Comment (Pinpickle) [inline Views/home.stencil]: `hx` tags are supposed to be headings of sections, but here (and I've seen this elsewhere), it's being used stylistically (`h4` is about the right size that you want).

We might want to make this the correct tag semantically (e.g. in a `p` element instead, but a `div` usually works just fine). Elsewhere you've used the `Text__headerScale` mixin which I think is a good way to achieve your desired convenience while also keeping things semantic.

You can also make a `Text--h4` modifier if you're really after some quick-to-hand convenience.

File: static/src/styles/core/color.styl [hunk 3]
```diff
@@ -1,12 +1,15 @@
 $Color__light = #F2F2F2;
+$Color__medium = #878787;
```
Comment (Pinpickle) [inline static/src/styles/core/color.styl]: What exactly is this a `medium` of? 🤔

**Comment by Pinpickle** [comment]
This is awesome! I left a couple of pretty small comments but otherwise this is pretty true to BEM methodology.

File: static/src/styles/core/color.styl (hunk 3 above)
Comment (jaredkhan) [inline static/src/styles/core/color.styl]: Any ideas for this? 'mediumGrey'? 'mediumBrightnessMuted'? 'fadedTextForLightBG'

Comment (Pinpickle) [inline static/src/styles/core/color.styl]: There isn't a great answer here, and we don't need to break our brains figuring it out so `mediumGray` sounds good to me (note the US spelling :cry:)

**Comment by Pinpickle** [comment]
:shipit:

**Pull request merged**